### PR TITLE
texlive: build full TEXMFROOT for non-relocated packages

### PR DESCRIPTION
**Pull request by xworld21**
###### Description of changes
The 'non-relocated' packages of TeX Live expect to live in `$TEXMFROOT/texmf-dist` (e.g. to search for perl modules). This change builds TEXMFROOT instead of just TEXMFDIST to accommodate those packages, and obviate the need for patching `fmtutil`.

The fact that `fmtutil` works without any patching doubles up as test, to make sure we don't mess this up in the future.

This unbreaks some other scripts with hardcoded `$TEXMFROOT/texmf-dist`, for instance `pedigree` (which otherwise complains with `compilation aborted`), and it should make `texlive` more robust against new scripts distributed by TeX Live.

PS: TEXMFROOT is also home to the `$TEXMFROOT/tlpkg` folder, so this PR prepares the ground for adding 'tlpkg' containers (in addition to run, doc, source).

PPS: as it happens, `pedigree` uses an hardcoded `$TEXMFROOT/texmf-config`, but that should *really* be `$TEXMFSYSCONFIG` and should probably be fixed upstream.

###### Things done
- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/typesetting/tex/texlive/combine.nix [hunk 1]
```diff
@@ -99,7 +103,10 @@ in (buildEnv {
     mkdir -p "$TEXMFCNF"
     if [ -e "$TEXMFDIST/web2c/texmfcnf.lua" ]; then
       sed \
+        -e "s,\(TEXMFOS[ ]*=[ ]*\)[^\,]*,\1\"$TEXMFROOT\",g" \
```
Comment (apfelkuchen6) [inline pkgs/tools/typesetting/tex/texlive/combine.nix:106]: What is the reason of the square brackets around the space? Do they do anything?

**Review by apfelkuchen6 — APPROVED** [review]
LGTM

File: pkgs/tools/typesetting/tex/texlive/combine.nix (hunk 1 above)
Comment (xworld21) [reply pkgs/tools/typesetting/tex/texlive/combine.nix:106]: I guess it used to be `[ \t]*`? I am not sure, really. Some code here is pretty old.

**Comment by veprbl** [comment]
@GrahamcOfBorg build tests.texlive

**Pull request merged**